src/hooks/useWifiInfo.ts

```diff
@@ -0,0 +1,12 @@
+import { useNetInfo } from "@react-native-community/netinfo";
+import { useCanAccessWifiSSID } from "./useCanAccessWifiSSID";
+
+export function useWifiInfo() {
+  const permissionEnabled = useCanAccessWifiSSID();
+  const net = useNetInfo({ shouldFetchWiFiSSID: true });
+  return {
+    permissionEnabled,
+    wifiConnected: net.type === "wifi" && net.isConnected,
+    ssid: net.type === "wifi" ? net.details.ssid : null,
```

Comment:
Needed to cover the following cases with this:

1. Wifi connected and SSID available
2. Wifi connected and SSID not available due to some unknown networking issue (not sure if this is likely)
3. Wifi connected and SSID not available due to user not providing relevant location permission that allows getting SSID
4. Wifi not connected and SSID not available